orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/TaskController.groovy

```diff
@@ -312,52 +313,63 @@ class TaskController {
   ) {
     validateSearchForPipelinesByTriggerParameters(triggerTimeStartBoundary, triggerTimeEndBoundary, startIndex, size)
 
-    final Map triggerParams = decodeTriggerParams(encodedTriggerParams) // Returned map will be empty if encodedTriggerParams is null
+    // Returned map will be empty if encodedTriggerParams is null
+    final Map triggerParams = decodeTriggerParams(encodedTriggerParams)
 
-    Set<String> triggerTypesAsSet = (triggerTypes && triggerTypes != "*") ? triggerTypes.split(",") as Set : null // null means all trigger types
-    Set<String> statusesAsSet = (statuses && statuses != "*") ? statuses.split(",") as Set : null // null means all statuses
+    Set<String> triggerTypesAsSet = (triggerTypes && triggerTypes != "*")
+      ? triggerTypes.split(",") as Set
+      : null // null means all trigger types
 
-    // Filter by application
-    List<String> pipelineConfigIds = application == "*" ?
-    getPipelineConfigIdsOfReadableApplications() :
-    front50Service.getPipelines(application, false)*.id as List<String>
-
-    List<Execution> pipelineExecutions = executionRepository.retrievePipelinesForPipelineConfigIdsBetweenBuildTimeBoundary(pipelineConfigIds, triggerTimeStartBoundary, triggerTimeEndBoundary, size)
-      .subscribeOn(Schedulers.io())
-      .filter{
-        // Filter by pipeline name
-        if (pipelineName && pipelineName != it.name) {
-          return false
+    // Filter by application (and pipeline name, if that parameter has been given in addition to application name)
+    List<String> pipelineConfigIds
+    if (application == "*") {
+      pipelineConfigIds = getPipelineConfigIdsOfReadableApplications()
+    } else {
+      List<Map<String, Object>> pipelines = front50Service.getPipelines(application, false)
+      pipelines = pipelines.stream().filter({ pipeline ->
+        if (pipelineName != null && pipelineName != "") {
+          return pipeline.get("name") == pipelineName
+        } else {
+          return true
         }
+      }).collect(Collectors.toList())
+      pipelineConfigIds = pipelines*.id as List<String>
+    }
+
+    ExecutionCriteria executionCriteria =  new ExecutionCriteria()
+      .setSortType( reverse ? REVERSE_BUILD_TIME : BUILD_TIME )
+    if (statuses != null && statuses != "") {
+      executionCriteria.setStatuses(statuses.split(",").toList())
```

Comment:
Hmm. I wonder if it'd be useful to have a `ExecutionCriteria setStatuses(String statuses)` that will use a splitter?

Reply:
It's easy enough to do at the call site, I don't think adding a method would provide a lot of value.